altos-rust/port/cortex-m0/src/peripheral/usart/mod.rs

```diff
@@ -1,123 +1,305 @@
-// peripheral/serial/mod.rs
-// AltOSRust
-//
-// Created by Daniel Seitz on 11/30/16
+/*
+ * Copyright © 2017 AltOS-Rust Team
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc.,
+ * 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA.
+ */
+
 mod control;
 mod defs;
 mod baudr;
 mod tdr;
+mod rdr;
 mod isr;
+mod icr;
+
+/// This module is the highest level in the Usart heirarchy for implementing
```

Comment:
Since none of the other modules are public, this is where all the documentation will be generated actually